src/js/tabs/firmware_flasher.js

```diff
@@ -119,132 +138,195 @@ TABS.firmware_flasher.initialize = function (callback) {
-                            return;
-                        }
-
-                        var target = match[2];
-                        var format = match[4];
-
-                        if (format != 'hex') {
-                            return;
-                        }
-
-                        var date = new Date(release.published_at);
-                        var formattedDate = ("0" + date.getDate()).slice(-2) + "-" + ("0"+(date.getMonth()+1)).slice(-2) + "-" + date.getFullYear() + " " + ("0" + date.getHours()).slice(-2) + ":" + ("0" + date.getMinutes()).slice(-2);
-
-                        var descriptor = {
-                            "releaseUrl": release.html_url,
-                            "name"      : version,
-                            "version"   : version,
-                            "url"       : asset.browser_download_url,
-                            "file"      : asset.name,
-                            "target"    : target,
-                            "date"      : formattedDate,
-                            "notes"     : release.body
-                        };
-                        releases[target].push(descriptor);
-                    });
-                });
-                var selectTargets = [];
-                Object.keys(releases)
-                    .sort()
-                    .forEach(function(target, i) {
-                        var descriptors = releases[target];
-                        descriptors.forEach(function(descriptor){
-                            if($.inArray(target, selectTargets) == -1) {
-                                selectTargets.push(target);
-                                var select_e =
-                                        $("<option value='{0}'>{0}</option>".format(
-                                                descriptor.target
-                                        )).data('summary', descriptor);
-                                boards_e.append(select_e);
-                            }
+        function loadUnifiedBuilds(builds) {
+            var expirationPeriod = 3600; // One of your earth hours.
+            var checkTime = Math.floor(Date.now()/1000); // Lets deal in seconds.
+            // unifiedTargets[0] is STM32F411
+            if (builds && builds[unifiedTargets[0]]) {
+                console.log('loaded some builds for later');
+                var storageTag = 'unifiedSourceCache';
+                chrome.storage.local.get(storageTag, function (result) {
+                    let storageObj = result[storageTag];
+                    if(!storageObj || !storageObj.lastUpdate || checkTime - storageObj.lastUpdate > expirationPeriod ) {
+                        console.log('go get', unifiedSource);
+                        $.get(unifiedSource, function(data,textStatus,jqXHR) {
+                            let newObj = {};
+                            newObj[storageTag]={};
```

Comment:
Readability and maintainability are improved heaps if you put spaces around operators, and after commas.